File: hw/santorum/u2/hw19Inheritance/Java/Farmsystem/src/main/java/ec/edu/espe/farmsystem/view/FarmSystem.java
package ec.edu.espe.farmsystem.view;

import ec.edu.espe.farmsystem.model.Cage;
import ec.edu.espe.farmsystem.model.Chicken;
import ec.edu.espe.farmsystem.model.Cow;
import ec.edu.espe.farmsystem.model.FarmAnimal;
import ec.edu.espe.farmsystem.model.Location;
import ec.edu.espe.farmsystem.model.Pig;
import ec.edu.espe.farmsystem.model.Sheep;
import java.util.Date;

/**
 *
 * @author Thais Santorum, Paradigm
 */


public class FarmSystem {
    public static void main(String[] args) {
        
        int xCoordinate = 10;
        int yCoordinate = 20;
        Location location = new Location(xCoordinate, yCoordinate);

        Cage cage = new Cage(1, "Corral for cows", 2);

        int id = 1;
        String breed = "Holstein";
        Date bornOn = new Date(2025, 2, 1);
        boolean isAbleToReproduce = false;
        float weight = 10.4F;
        String gender = "male";

   
        FarmAnimal chicken = new Chicken(true, 0, id, breed, bornOn, gender, isAbleToReproduce, weight, cage);
        System.out.println("Animal --> " + chicken);


        FarmAnimal cow = new Cow(true, 20.5F, 2, "Jersey", bornOn, "female", true, 450.0F, cage);
        System.out.println("Animal --> " + cow);

        
        FarmAnimal sheep = new Sheep(new Date(), 3, "Merino", bornOn, "female", true, 60.0F, cage);
        System.out.println("Animal --> " + sheep);


        FarmAnimal pig = new Pig(3.5F, false, 4, "Landrace", bornOn, "male", false, 120.0F, cage);
        System.out.println("Animal --> " + pig);

    }
}

File: hw/santorum/u2/hw19Inheritance/JavaS/view/FarmSystem.js
import { Location } from "../model/Location.js";
import { Cage } from "../model/Cage.js";
import { Chicken } from "../model/Chicken.js";
import { Cow } from "../model/Cow.js";
import { Sheep } from "../model/Sheep.js";
import { Pig } from "../model/Pig.js";

const location = new Location(10, 20);
const cage = new Cage(1, "Corral for cows", 2);

const bornOn = new Date(2025, 1, 1); // Feb 1 2025


const chicken = new Chicken(true, 0, 1, "Holstein", bornOn, "male", false, 10.4, cage);
console.log("Animal -->", chicken.toString());


const cow = new Cow(true, 20.5, 2, "Jersey", bornOn, "female", true, 450, cage);
console.log("Animal -->", cow.toString());


const sheep = new Sheep(new Date(), 3, "Merino", bornOn, "female", true, 60, cage);
console.log("Animal -->", sheep.toString());


const pig = new Pig(3.5, false, 4, "Landrace", bornOn, "male", false, 120, cage);
console.log("Animal -->", pig.toString());
